src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,45 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health) {
+    this.name = name;
+    this.health = health || 100;
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health) {
+    super(name, health);
+
+    this.hidden = false;
+    Animal.alive.push(this);
+  }
+
+  hide() {
+    this.hidden = !this.hidden;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health) {
+    super(name, health);
+
+    Animal.alive.push(this);
+  }
+
+  bite(animal) {
+    if (!animal.hidden && animal.__proto__ !== Carnivore.prototype) {
+      animal.health -= 50;
+    }
+
+    if (!animal.health) {
+      const index = Animal.alive.indexOf(animal);
+
+      Animal.alive.splice(index, 1);
+    }
```

Comment:
there are two potential things:

- your if will only work for health = 0; what if health becomes less than 0? 
- if the findIndex method does not find the item (for some reason), it will return -1 and splice with -1 will remove the last element. this is not likely to happen, but it is something to think about